jumpcloud_bootstrap_template.sh

```diff
@@ -1196,21 +1598,153 @@ if [[ -f $DEP_N_GATE_DONE ]]; then
+                echo "$(date "+%Y-%m-%d %H:%M:%S"): Error during JumpCloud Command Result Lookup" >>"$DEP_N_DEBUG"
+                break
+            fi
+        done
+
+        ## End Wait for command results
+        ## Begin parse results
+        err='Error'
+        # err='Error:-[0-9]+'
+        if [[ $commandResponseContent =~ $err ]]; then
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): [ERROR] sysadminctl encountered a general error, the $user1 && $user2 users may not be deleted from the system" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): The following errors were returned when attempting to delete the users:" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): ################################  Error Details ################################" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): $commandResponseContent" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): ################################################################################" >>"$DEP_N_DEBUG"
         fi
+        errorNF='not found'
+        # errorNF='User \w+ not found'
+        if [[ $commandResponseContent =~ $errorNF ]]; then
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): [ERROR] sysadminctl encountered a user not found error, the $user1 && $user2 users may not be deleted from the system" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): The following errors were returned when attempting to delete the users:" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): ################################  Error Details ################################" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): $commandResponseContent" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): ################################################################################" >>"$DEP_N_DEBUG"
+        fi
+        ## end parse results
 
+        ## delete Command:
+        cmdCreateRequest=$(
+            curl -s -o $DEP_N_HTTP_RESPONSE -w "%{http_code}" \
+            -X DELETE \
+            -H 'Accept: application/json' \
+            -H 'Content-Type: application/json' \
+            -H "x-api-key: $APIKEY" \
+                --url https://console.jumpcloud.com/api/commands/$commandID
+        )
+        responseContent=$(cat $DEP_N_HTTP_RESPONSE)
+        if [[ $cmdCreateRequest != "200" ]]; then
+            # handle error
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): [error] HTTP Response does not indicate success" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): HTTP Response: $responseContent" >>"$DEP_N_DEBUG"
+        else
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): Command used to trigger the user cleanup was removed" >>"$DEP_N_DEBUG"
+        fi
+        # Wait 10 seconds for user tables to update
+        echo "$(date "+%Y-%m-%d %H:%M:%S"): Wait 30s" >>"$DEP_N_DEBUG"
+        sleep 30
```

Comment:
We can have a race condition here when the command results are seen by the response above but the command hasn't finished executing yet. I've just seen that once but I'm adding a sleep 30 here to account for it.